Source/main.cpp

```diff
@@ -347,10 +347,10 @@ void main_main ()
     // copy new solution into old solution
     for(int comp = 0; comp < 3; comp++)
     {
-       MultiFab::Copy(Mfield_old[comp], Mfield[comp], 0, 0, 3, 1);
-       MultiFab::Copy(Mfield_pre[comp], Mfield[comp], 0, 0, 3, 1);
-       MultiFab::Copy(Mfield_prev_iter[comp], Mfield[comp], 0, 0, 3, 1);
-       MultiFab::Copy(Mfield_error[comp], Mfield[comp], 0, 0, 3, 1);
+       MultiFab::Copy(Mfield_old[comp], Mfield[comp], 0, 0, 3, Mfield[comp].nGrow());
+       MultiFab::Copy(Mfield_pre[comp], Mfield[comp], 0, 0, 3, Mfield[comp].nGrow());
+       MultiFab::Copy(Mfield_prev_iter[comp], Mfield[comp], 0, 0, 3, Mfield[comp].nGrow());
+       MultiFab::Copy(Mfield_error[comp], Mfield[comp], 0, 0, 3, Mfield[comp].nGrow());
```

Comment:
If you REALLY want to be a memory minimalist, I think Mfield_error requires 0 ghost cells since we only compute the error in the valid region.

Reply:
fixed